plugins/plugin-traefik/plugin-traefik-docker/src/main/java/org/eclipse/che/plugin/traefik/TraefikCreateContainerInterceptor.java

```diff
@@ -0,0 +1,148 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2017 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.plugin.traefik;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.inject.name.Named;
+
+import org.aopalliance.intercept.MethodInterceptor;
+import org.aopalliance.intercept.MethodInvocation;
+import org.eclipse.che.commons.annotation.Nullable;
+import org.eclipse.che.plugin.docker.client.DockerConnector;
+import org.eclipse.che.plugin.docker.client.json.ContainerConfig;
+import org.eclipse.che.plugin.docker.client.json.ImageInfo;
+import org.eclipse.che.plugin.docker.client.params.CreateContainerParams;
+import org.eclipse.che.plugin.docker.client.params.InspectImageParams;
+import org.eclipse.che.plugin.docker.machine.CustomServerEvaluationStrategy;
+import org.eclipse.che.plugin.docker.machine.ServerEvaluationStrategy;
+import org.eclipse.che.plugin.docker.machine.ServerEvaluationStrategyProvider;
+
+import javax.inject.Inject;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static java.lang.String.format;
+
+/**
+ * Intercept the calls on createContainer on docker Connector.
+ *
+ * @author Florent Benoit
+ */
+public class TraefikCreateContainerInterceptor implements MethodInterceptor {
+
+    /**
+     * Inject the server evaluation strategy provider.
+     */
+    private ServerEvaluationStrategyProvider serverEvaluationStrategyProvider;
+
+    /**
+     * Template.
+     */
+    private String template;
+
+
+    /**
+     * Grab labels of the config and from image to get all exposed ports and the labels defined if any
+     *
+     * @param methodInvocation
+     *         intercepting data of createContainer method on {@link DockerConnector}
+     * @return the result of the intercepted method
+     * @throws Throwable
+     *         if there is an exception
+     */
+    @Override
+    public Object invoke(MethodInvocation methodInvocation) throws Throwable {
+        ServerEvaluationStrategy serverEvaluationStrategy = serverEvaluationStrategyProvider.get();
+        // Abort if custom server evaluation strategy is not enabled.
+        if (!(CustomServerEvaluationStrategy.class.isInstance(serverEvaluationStrategy))) {
```

Comment:
in Guice module, we do not bind the interceptor if we do not enabled traefik.
https://github.com/eclipse/che/blob/683b6f4b8253cc83c8ce7500020c7bf2ee8a8130/plugins/plugin-traefik/plugin-traefik-docker/src/main/java/org/eclipse/che/plugin/traefik/TraefikDockerModule.java#L35-L40
so it's an extra check in case the interceptor is bound but the strategy is not the expected one.

And you can still enable the custom strategy without enabling the traefik plug-in/interceptor